### PR TITLE
feat: Amortize the cost of database setup in postgres through the use of template databases.

### PR DESCRIPTION
**Pull request by DanCardin**
Today we create a brand new test database for each test, it's completely empty, **and then** we populate the database with `ordered_actions` to fill it with the DDL/data we need in the tests.

The idea is that we make use of postgres' [template database](https://www.postgresql.org/docs/current/manage-ag-templatedbs.html) feature which replicates the state of a database from another database (which happens always, just your default template `template1` is empty by default). We avoid tons of python/sqlalchemy stuff, and presumably postgres is optimized to do what it does quickly.

Large test suite 
* **Before**: 1204 passed, 3 errors in 1458.88s (**0:24:18**) (funnily, i got errors on the head version, didn't look into why)
* **After (Statements, as-is)**: 1207 passed in 433.54s (**0:07:13**)
* **After (Statements -> StaticStatements)**: 1207 passed in 383.83s (**0:06:23**)

The above is with `-n 4` parallelism. the numbers get better as you increase parallelism. I stopped the original suite after 40 mins without parallelism, whereas with this PR it took 17m (faster than the parallelized pre-PR version)

For pytest-mock-resources' own test suite, it's essentially the same or slightly slower overall because there necessarily tends to be close to a 1-to-1 correspondence between each test and its fixture to test PMR's fixture args/features.

### CODE REVIEW

File: tests/fixture/database/test_ordered_actions.py [hunk 1]
```diff
@@ -35,19 +35,20 @@ class Object(Base):
...
 
 row_dependant_statements = Statements(
-    "CREATE TEMP TABLE user1 as SELECT DISTINCT CONCAT(name, 1) as name FROM stuffs.user"
+    "CREATE TABLE user1 as SELECT DISTINCT CONCAT(name, 1) as name FROM stuffs.user"
 )
 
 additional_rows = Rows(User(name="Perrier"), User(name="Mug"))
 
 
 def session_function(session):
```
Comment (DanCardin) [inline tests/fixture/database/test_ordered_actions.py:44]:
```
def foo(pg: Engine):
   pg.execute('insert foo')  # this should auto-commit

pg = create_postgres_fixture(foo)

def test_foo(pg: Engine):
   ...
```

```
def foo(pg: Session):
   pg.add(User(...)) # make this work without `pg.commit()`

pg = create_postgres_fixture(foo, session=True)

def test_foo(pg: Session):
   ...
```

Document this!

File: tests/fixture/database/test_ordered_actions.py [hunk 2]
```diff
@@ -35,19 +35,20 @@ class Object(Base):
 rows = Rows(User(name="Harold"), User(name="Gump"))
 
 row_dependant_statements = Statements(
-    "CREATE TEMP TABLE user1 as SELECT DISTINCT CONCAT(name, 1) as name FROM stuffs.user"
```
Comment (DanCardin) [inline tests/fixture/database/test_ordered_actions.py:38]: this shouldn't need to change, fix

**Comment by luke-mino-altherr** [comment]
i'm assuming youll fix tests?

**Comment by DanCardin** [comment]
yea, it was all passing a few moments ago, i added a new test and broke some stuff.

File: docs/source/relational/ordered-actions.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+   metadata were passed in immediately preceding the ``Rows`` action.
+
+
+Statements/StaticStatements
+---------------------------
+Either a :class:`Statements` or :class:`StaticStatements` object can be constructed,
+which will execute arbitrary SQL before entering the test.
+
+Both operate in exactly the same way, however a :class:`StaticStatements` is a
+sentinel to the library that the Statement is safe to amortize with
```
Comment (oakhan3) [inline docs/source/relational/ordered-actions.rst:170]:
```suggestion
Both operate in exactly the same way, however :class:`StaticStatements` let the
library know that the included SQL statements are a safe to "cache" in order to reduce database creation costs.
```

File: tests/test_non_session_container.py [hunk 4]
```diff
@@ -5,5 +5,5 @@
 
 @pytest.fixture(scope="session")
 def pmr_postgres_config():
-    print('asdfasdfasdf')
+    print("asdfasdfasdf")
```
Comment (oakhan3) [inline tests/test_non_session_container.py:8]: rm?

**Review by oakhan3 — APPROVED** [review]

**Pull request merged**